generate/base.py

```diff
@@ -49,7 +49,7 @@ def sample(logits: torch.Tensor, temperature: float = 1.0, top_k: Optional[int]
 def next_token(model: GPT, input_pos: torch.Tensor, x: torch.Tensor, **kwargs: Any) -> torch.Tensor:
     logits = model(x, input_pos)
     next = sample(logits, **kwargs)
-    return next.type_as(x)
```

Comment:
Turns out that `type_as` will also change the device. PyTorch has soft-deprecated it for this reason.